### PR TITLE
Fastboot compatibility

### PR DESCRIPTION
**Pull request by ronco**
Manually construct head component (for fastboot).

Instantiating the component automatically through the container throws an exception under fastboot because a new DomHelper is created which relies upon `document`.

For full functionality we will also need an index.html [placeholder](https://github.com/ronco/ember-cli-fastboot/commit/d6ef0f6cb7f128b365fcb8d44c998ac141a78f9b), and [updates to ember-fastboot-server](https://github.com/ronco/ember-fastboot-server/commit/9087bb5d324a8c6573d720e1e52f3f82727f69c2) to serialize the head component.

### CODE REVIEW

File: addon/instance-initializers/head.js [hunk 1]
```diff
@@ -1,9 +1,15 @@
 export function initialize(instance) {
   const container = instance.lookup ? instance : instance.container;
-  const component = container.lookup(
-    'component:head-content'
-  );
-  component.appendTo('head');
+  const componentFactory =
+          instance.__container__.lookupFactory('component:head-content');
```
Comment (rwjblue) [inline addon/instance-initializers/head.js]: use `instance._lookupFactory` instead

Comment (ronco) [inline addon/instance-initializers/head.js]: :+1:

File: addon/instance-initializers/head.js [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+  const renderer = container.lookup('renderer:-dom');
+  // explicitly set renderer & domhelper in case we're in fastboot
+  const component = componentFactory.create({
+    renderer,
+    container,
```
Comment (rwjblue) [inline addon/instance-initializers/head.js]: This isn't quite right in 2.3.0+.

Would need to do:

```
const component = componentFactory.create(
  instance.ownerInjection(), 
  {
    renderer,
    _domHelper: renderer._dom
  }
});
```

**Comment by rwjblue** [comment]
Left a few nit-picks, but looks good otherwise.

**Comment by ronco** [comment]
Thanks! It'll be a couple days till I can work on this again, but I'll try
to have it all buttoned up before next weeks hangout.

**Pull request merged**